packages/story-editor/src/utils/storyPageToNode.js

```diff
@@ -0,0 +1,176 @@
+  ref
+) {
+  return (
+    <ThemeProvider theme={ds_theme}>
+      <FontProvider>
+        <TransformProvider>
+          <UnitsProvider
+            pageSize={{
+              width,
+              height,
+            }}
+          >
+            <Page
+              ref={ref}
+              height={renderFullHeightThumb ? containerHeight : height}
+              width={width}
+            >
+              <PreviewWrapper background={page.backgroundColor}>
+                <FullHeight
+                  yOffset={
+                    renderFullHeightThumb ? (containerHeight - height) / 2 : 0
+                  }
+                >
+                  {page.elements.map((element) => (
+                    <DisplayElement
+                      key={element.id}
+                      previewMode
+                      element={element}
+                    />
+                  ))}
+                </FullHeight>
+              </PreviewWrapper>
+            </Page>
+          </UnitsProvider>
+        </TransformProvider>
+      </FontProvider>
+    </ThemeProvider>
+  );
+});
+
+/**
+ * @typedef {import('../../../types').Page} Page
+ */
+
+/**
+ * Takes a story page and generates a DOM node containing the rendered
+ * page. Returns a tuple containing the page DOM node and a cleanup
+ * method to remove the returned DOM node.
+ *
+ * **IMPORTANT:** Not calling the returned `cleanup()` method after use of
+ * page DOM node will result in memory leak.
+ *
+ * @param {Page} page Page object.
+ * @param {number} width desired width of image. Dictates height and container height
+ * @param {{ renderFullHeightThumb: boolean }} opts - options to alter the rendered node.
+ * @return {[HTMLElement, Function]} tuple containing DOM node and cleanup method
+ */
+async function storyPageToNode(page, width, opts = {}) {
+  const { renderFullHeightThumb = false } = opts;
+  const height = width * (1 / PAGE_RATIO);
+  const containerHeight = width * (1 / FULLBLEED_RATIO);
+
+  const bufferRoot = document.createElement('div');
+  bufferRoot.style.cssText = `
+     contain: strict;
+     position: absolute;
+     top: 0;
+     left: 0;
+     width: ${width}px;
+     height: ${containerHeight}px;
+     opacity: 0;
+     transform: translate(-100%, -100%);
+     pointer-events: none;
+   `;
+
+  const node = await new Promise((resolve) => {
+    const resolverRef = (htmlNode) => {
+      if (!htmlNode) {
+        return;
+      }
+      resolve(htmlNode);
+    };
+
+    render(
+      <PageWithDependencies
+        ref={resolverRef}
+        page={page}
+        width={width}
+        height={height}
+        containerHeight={containerHeight}
+        renderFullHeightThumb={renderFullHeightThumb}
+      />,
+      bufferRoot
+    );
+  });
+
+  document.body.appendChild(bufferRoot);
+  const cleanup = () => {
+    unmountComponentAtNode(bufferRoot);
+    document.body.removeChild(bufferRoot);
+  };
+
+  return [node, cleanup];
+}
+export default storyPageToNode;
```

Comment:
pulled out directly from `storyPageToDataUrl` so we can use it for both `storyPageToDataUrl`  & `storyPageToCanvas`. didn't change the behavior at all beyond exposing the `renderFullHeightThumb` prop so that the generated node could be full bleed or not